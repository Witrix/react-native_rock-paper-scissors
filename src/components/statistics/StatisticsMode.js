import React, { Component } from 'react';
import {
    View,
    Text,
    StyleSheet,
} from 'react-native';
import StatisticsBar    from "./StatisticsBar";
import AppColors        from "../../themes/colors";
import Spacer           from "../general/Spacer";
import MyLocalization   from "../../my-lib/MyLocalzation";

export default class StatisticsMode extends Component {

    constructor(props) {
        super(props);
        this.state = {
            stat: props.stat,
        }
    }

    componentWillReceiveProps(nextProps) {
        this.setState({
            stat: nextProps.stat,
        })
    }

    render() {
        if (this.state.stat) {
            let numberVictory = this.state.stat.victory || 0;
            let numberDefeat = this.state.stat.defeat || 0;
            let numberEquality = this.state.stat.equality || 0;
            let numberTotal = (numberDefeat + numberVictory + numberEquality) || 1;
            let percentVictory = numberVictory / numberTotal;
            let percentDefeat = numberDefeat / numberTotal;
            let percentEquality = numberEquality / numberTotal;
            return (
                <View>
                    <Text style={style.text}>{MyLocalization.getString('victory')}: {Math.round(percentVictory * 100)}% ({numberVictory})</Text>
                    <StatisticsBar percent={percentVictory} color={AppColors.positive}/>
                    <Spacer size={5}/>
                    <Text style={style.text}>{MyLocalization.getString('defeat')}: {Math.round(percentDefeat * 100)}% ({numberDefeat})</Text>
                    <StatisticsBar percent={percentDefeat} color={AppColors.brand.secondary}/>
                    <Spacer size={5}/>
                    <Text style={style.text}>{MyLocalization.getString('equality')}: {Math.round(percentEquality * 100)}% ({numberEquality})</Text>
                    <StatisticsBar percent={percentEquality} color={AppColors.brand.primary}/>
                </View>
            );
        } else {
            return (
                <View style={style.noData}>
                    <Text>{MyLocalization.getString('noDataForStat')}</Text>
                </View>
            );
        }
    }
}

const style = StyleSheet.create({
    text: {
        color: 'black',
    },
    noData: {
        alignItems: 'center',
        justifyContent: 'center'
    }
});
