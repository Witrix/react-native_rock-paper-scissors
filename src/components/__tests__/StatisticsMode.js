import 'react-native';
import React                    from 'react';
import StatisticsMode           from '../statistics/StatisticsMode';
import renderer                 from 'react-test-renderer';

it('Render correctly', () => {
  const tree = renderer.create(
    <StatisticsMode />
  );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: valid props', () => {
    const tree = renderer.create(
        <StatisticsMode stat={{victory: 10, defeat: 10, equality: 10}}/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: invalid props', () => {
    const tree = renderer.create(
        <StatisticsMode stat={{a: 'b'}}/>
    );
    expect(tree).toMatchSnapshot();
});