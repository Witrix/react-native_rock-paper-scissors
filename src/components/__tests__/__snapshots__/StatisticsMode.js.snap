// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    No data available at the moment.
  </Text>
</View>
`;

exports[`Render correctly: invalid props 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
      }
    }
  >
    Victory
    : 
    0
    % (
    0
    )
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "margin": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#AED581",
          "flex": 0.01,
          "height": 10,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 0.99,
          "height": 10,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 5,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
      }
    }
  >
    Defeat
    : 
    0
    % (
    0
    )
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "margin": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFB74D",
          "flex": 0.01,
          "height": 10,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 0.99,
          "height": 10,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 5,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
      }
    }
  >
    Equality
    : 
    0
    % (
    0
    )
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "margin": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#29B6F6",
          "flex": 0.01,
          "height": 10,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 0.99,
          "height": 10,
        }
      }
    />
  </View>
</View>
`;

exports[`Render correctly: valid props 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
      }
    }
  >
    Victory
    : 
    33
    % (
    10
    )
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "margin": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#AED581",
          "flex": 0.3333333333333333,
          "height": 10,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 0.6666666666666667,
          "height": 10,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 5,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
      }
    }
  >
    Defeat
    : 
    33
    % (
    10
    )
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "margin": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFB74D",
          "flex": 0.3333333333333333,
          "height": 10,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 0.6666666666666667,
          "height": 10,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 5,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
      }
    }
  >
    Equality
    : 
    33
    % (
    10
    )
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "margin": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#29B6F6",
          "flex": 0.3333333333333333,
          "height": 10,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 0.6666666666666667,
          "height": 10,
        }
      }
    />
  </View>
</View>
`;
